zulip_bots/zulip_bots/run.py

```diff
@@ -24,31 +45,61 @@ def parse_args():
         See lib/readme.md for more context.
         '''
 
-    parser = optparse.OptionParser(usage=usage)
-    parser.add_option('--quiet', '-q',
-                      action='store_true',
-                      help='Turn off logging output.')
+    parser = argparse.ArgumentParser(usage=usage)
+    parser.add_argument('name',
+                        action='store',
+                        nargs='?',
+                        default=None,
+                        help='the name of an existing bot to run')
+
+    parser.add_argument('--quiet', '-q',
+                        action='store_true',
+                        help='Turn off logging output.')
 
-    parser.add_option('--config-file',
-                      action='store',
-                      help='(alternate config file to ~/.zuliprc)')
+    parser.add_argument('--config-file',
+                        action='store',
+                        help='(alternate config file to ~/.zuliprc)')
 
-    parser.add_option('--force',
-                      action='store_true',
-                      help='Try running the bot even if dependencies install fails.')
-    (options, args) = parser.parse_args()
-    if not args:
-        parser.error('You must specify the name of the bot!')
+    parser.add_argument('--path-to-bot',
+                        action='store',
+                        help='path to the file with the bot handler class')
 
-    return (options, args)
+    parser.add_argument('--force',
+                        action='store_true',
+                        help='Try running the bot even if dependencies install fails.')
+
+    parser.add_argument('--provision',
+                        action='store_true',
+                        help='Install dependencies for the bot.')
+    options = parser.parse_args()
+
+    if not options.name and not options.path_to_bot:
+        error_message = """
+You must either specify the name of an existing bot or
+specify a path to the file (--path-to-bot) that contains
+the bot handler class.
+"""
+        parser.error(error_message)
+
+    return options
 
 
 def main():
     # type: () -> None
-    (options, args) = parse_args()
-    bot_name = args[0]
+    options = parse_args()
+
+    if options.path_to_bot:
+        if options.provision:
+            bot_dir = os.path.dirname(os.path.abspath(options.path_to_bot))
+            provision_bot(bot_dir, options.force)
+        lib_module = import_module_from_source(options.path_to_bot, name=options.name)
+    else:
```

Comment:
You can probably use `elif` for now here.  A short comment explaining why these options are mutually exclusive would be helpful, too.